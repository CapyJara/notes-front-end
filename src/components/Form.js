import React from 'react';
import PropTypes from 'prop-types';

function NoteDForm({ title, body, handleSubmit, handleChange }) {

  return (
    <form onSubmit={handleSubmit} style={formStyle}>
      <button style={buttonStyle}>Submit</button>
      <input style={titleInput} type="text" name="title" value={title} placeholder={title || 'Note Title'} onChange={handleChange} />
      <textarea style={bodyInput} name="body" value={body} placeholder={body || 'Note Body'} onChange={handleChange} />
    </form>
  );

}

NoteDForm.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default NoteDForm;

const formStyle = {
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'start'
};

const buttonStyle = {
  width: '80px',
  margin: '10px'
};

const titleInput = {
  width: '225px',
  textAlign: 'center',
  margin: '10px'
};

const bodyInput = {
  height: '200px',
  width: '400px',
  margin: '10px'
};
