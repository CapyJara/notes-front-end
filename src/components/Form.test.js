import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('renders a Form', () => {
    const title = '';
    const body = '';
    const handleChange = jest.fn();
    const handleSubmit = jest.fn();

    const wrapper = shallow(<Form title={title} body={body} handleChange={handleChange} handleSubmit={handleSubmit} />);
    expect(wrapper).toMatchSnapshot();
  });
});
