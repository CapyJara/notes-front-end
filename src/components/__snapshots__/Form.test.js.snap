// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders a Form 1`] = `
<form
  onSubmit={[MockFunction]}
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "start",
    }
  }
>
  <button
    style={
      Object {
        "margin": "10px",
        "width": "80px",
      }
    }
  >
    Submit
  </button>
  <input
    name="title"
    onChange={[MockFunction]}
    placeholder="Note Title"
    style={
      Object {
        "margin": "10px",
        "textAlign": "center",
        "width": "225px",
      }
    }
    type="text"
    value=""
  />
  <textarea
    name="body"
    onChange={[MockFunction]}
    placeholder="Note Body"
    style={
      Object {
        "height": "200px",
        "margin": "10px",
        "width": "400px",
      }
    }
    value=""
  />
</form>
`;
